src/entt/core/memory.hpp

```diff
@@ -280,8 +290,25 @@ constexpr Type make_obj_using_allocator(const Allocator &allocator, Args &&...ar
  * @return A pointer to the newly created object of the given type.
  */
 template<typename Type, typename Allocator, typename... Args>
-constexpr Type *uninitialized_construct_using_allocator(Type *value, const Allocator &allocator, Args &&...args) {
-    return std::apply([value](auto &&...curr) { return new(value) Type(std::forward<decltype(curr)>(curr)...); }, internal::uses_allocator_construction<Type>::args(allocator, std::forward<Args>(args)...));
+constexpr Type *uninitialized_construct_using_allocator(Type *value, Allocator &allocator, Args &&...args) {
```

Comment:
This change should be semantically equal for the default allocator but it makes use of the `construct_at` provided by a specialized allocator.

Reply:
There is a compile-time  branch now. How can it be semantically equal? I'm pretty sure one can easily pack an example that works in one case and behaves differently in the other one. The whole change would be pointless otherwise. 🤔